bootstrap/pkg/kfapp/kind/kind.go

```diff
@@ -0,0 +1,165 @@
+	"sigs.k8s.io/kind/pkg/util"
+)
+
+type Kind struct {
+	KfDef       *kfdefs.KfDef
+	Environment string
+}
+
+type KindClusterPhase string
+
+const (
+	KindPluginName = kftypesv3.Kind
+)
+
+func GetPlatform(kfdef *kfdefs.KfDef) (kftypesv3.Platform, error) {
+	_kindPlatform := &Kind{
+		KfDef:       kfdef,
+		Environment: "test", // or development
+	}
+	return _kindPlatform, nil
+}
+
+func (pluginSpec *KindPluginSpec) IsValid() (bool, error) {
+
+	defaultSC := pluginSpec.DefaultStorageClass == ""
+	if defaultSC {
+		return false, errors.New("default storage class not set")
+	}
+	return true, nil
+}
+
+// Init creates the KinD cluster and sets the KUBECONFIG
+func (kind *Kind) Init(resources kftypesv3.ResourceEnum) error {
+	fmt.Printf("Kind Plugin Init test...\n")
+	// Check for system requirements - docker daemon etc...
+	return nil
+}
+
+func (kind *Kind) Generate(resources kftypesv3.ResourceEnum) error {
+	if err := kind.KfDef.SyncCache(); err != nil {
+		log.Errorf("Unable to sync cache: %v", err)
+		return errors.WithStack(err)
+	}
+
+	pluginSpec := &KindPluginSpec{}
+	err := kind.KfDef.GetPluginSpec(KindPluginName, pluginSpec)
+	if err != nil {
+		return &kfapis.KfError{
+			Code:    int(kfapis.INVALID_ARGUMENT),
+			Message: fmt.Sprintf("Unable to get Kind plugin spec: %v", err),
+		}
+	}
+	pluginSpec.DefaultStorageClass = "standard"
+	fmt.Printf("Kind Plugin Generate test UseBasicAuth set to: %v\n", pluginSpec.UseBasicAuth)
+	return nil
+}
+
+func (kind *Kind) Apply(resources kftypesv3.ResourceEnum) error {
+	fmt.Printf("Kind Plugin Apply test...\n")
+	err := CreateKindCluster(kind.KfDef.Name, kind)
+	if err != nil {
+		return &kfapis.KfError{
+			Code:    int(kfapis.INTERNAL_ERROR),
+			Message: fmt.Sprintf("Unable to create KinD cluster: %v", err),
+		}
+	}
+
+	return nil
+}
+
+func (kind *Kind) Delete(resources kftypesv3.ResourceEnum) error {
+	fmt.Printf("Kind Plugin Delete test...")
+	err := DeleteKindCluster(kind.KfDef.Name)
+	if err != nil {
+		return &kfapis.KfError{
+			Code:    int(kfapis.INTERNAL_ERROR),
+			Message: fmt.Sprintf("Could not delete KinD cluster: %v", err),
+		}	
+	}
+	return nil
+}
+
+func CreateKindCluster(clusterName string, kind *Kind) error {
+	// Check if the cluster name already exists
+	known, err := cluster.IsKnown(clusterName)
+	if err != nil {
+		return err
+	}
+	if known {
+		log.Infof("Skipping creating a cluster with the name %v, because it already exists", clusterName)
+		return nil
+	}
+
+	// Create a context for the cluster and validate it
+	ctx := cluster.NewContext(clusterName)
+	KubeconfigPath := ctx.KubeConfigPath()
+	log.Infof("Creating KinD cluster with the kubeconfig placed at: %v", KubeconfigPath)
+
+	kindManifestsDir := kind.KfDef.Status.ReposCache["manifests"].LocalPath + "/kind"
+	kindConfigFilePath := kindManifestsDir + "/kind-config.yaml"
+	// KinD Node Image version
+	imageName := "kindest/node:v1.15.0@sha256:b4d092fd2b507843dd096fe6c85d06a27a0cbd740a0b32a880fe61aba24bb478"
```

Comment:
Question: Is it better to take this also from kind-config.yaml?

Reply:
Thanks for the review @nrchakradhar, yes it will be. Added.